### PR TITLE
Add insertion sort explanation + corrected grammar

### PR DESCRIPTION
**Pull request by Leoriem-code**
Like the title said.
I really need to proof-read myself before pushing changes

### CODE REVIEW

File: fr/filtres_audio/filtre_butterworth.md [hunk 1]
```diff
@@ -4,10 +4,10 @@
...
+* Le filtre de Butterworth est originellement un filtre passe-bas, et c'est le filtre le plus utilisé du fait de son gain extrêmement plat dans la zone passe-bande.  
+* L'avantage d'un tel filtre est d'amplifier toutes les fréquences voulues de façon uniforme, au lieu d'amplifier certaines fréquences plus que d'autre comme le faisaient les filtres en usage avant sa découverte (les filtres [elliptiques](https://fr.wikipedia.org/wiki/Filtre_elliptique) et [de Tchebychev](https://fr.wikipedia.org/wiki/Filtre_de_Tchebychev)).
 * De plus, Butterworth a prouvé que le filtre passe-bas pouvait être modifié facilement afin de donner des filtres passe-haut, passe-bande ou coupe-bande.
-* La fonction de transfert d'un filtre de Butterworth passe-bas d'ordre $n$ dans le domaine fréquentielle est:
+* La fonction de transfert d'un filtre de Butterworth passe-bas d'ordre $n$ dans le domaine fréquentielle est :
```
[the remark this answers is not in the document]
Comment (Leoriem-code) [reply fr/filtres_audio/filtre_butterworth.md:10]: In French, there is a space here

[the remark this answers is not in the document]
Comment (Panquesito7) [reply fr/filtres_audio/filtre_butterworth.md:10]: > In French, there is a space here

Okay, searched properly; that's correct. 🙂

**Review by Panquesito7 — DISMISSED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍

File: fr/tri/Tri_par_insertion.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Le principe du tri par insertion est de parcourir la liste de valeurs, en les classant par rapport aux valeurs déjà parcourues.  
+On trie la 2ème valeur par rapport à la 1ère, la 3ème par rapport à la 2ème et la 1ème, etc.
+
+### Complexités
+
+Lors de l'évaluation des complexités d'un algorithme de tri, il est nécessaire de prendre en compte non seulement le cas général, mais également le meilleur et le pire cas.
+
+* La complexité spatiale d'un tri par insertion est simple et est constante à travers les cas. Il s'agit de la liste des valeurs, car le tri est effectué directement dans la liste, plus une variable qui servira à retenir la valeur que l'on est en train de trier.
+* La complexité temporelle varie beaucoup selon les cas :
```
Comment (appgurueu) [inline fr/tri/Tri_par_insertion.md:15]: Perhaps elaborate on:
* Which input constitutes the worst case / best case / average case (list of numbers sorted in reverse order / sorted in order / average permutation)?
* How is the complexity obtained? (having to shift all sorted elements when inserting in the worst case)

**Review by appgurueu — COMMENTED** [review]
Consider adding a section on alternative, more efficient sorting algorithms.

File: fr/tri/Tri_par_insertion.md (hunk 2 above)
Comment (Leoriem-code) [reply fr/tri/Tri_par_insertion.md:15]: I will, thanks

File: fr/tri/Tri_par_insertion.md [hunk 3]
```diff
@@ -34,12 +37,20 @@ Dans notre exemple, on va trier la liste $84, 33, 82, 53, 50, 66, 10, 92, 96, 77
...
+![Tri par insertion dans le pire cas](../Exemples/.gif/TriInsertionPire.gif)
 
 ## Implémentation
 
 * [Python](https://github.com/TheAlgorithms/Python/blob/master/sorts/insertion_sort.py)
+
+## Pour aller plus loin
+
+* Dans le meilleur cas, nous n'effectuons qu'une unique comparaison. En effet, le terme comparé et plus grand que le terme juste avant, donc aucune modification n'est à effectuer. (ce qui correspond bien à $n$ opérations)
+* Dans le pire des cas, on doit au contraire "redescendre" tous les termes un par un. On effectue une opération pour remonter le 2ème terme, deux pour le 3ème, etc. On effectue donc au total $\sum{k=0}{n}{k}$ opération, soit bien un ordre de $N^2/2$ (ou plus précisément $N^2$)
```
Comment (appgurueu) [inline fr/tri/Tri_par_insertion.md:57]: $N^2/2$ is the more precise term (which is $\Theta(N^2)$ OFC), obtained through applying the [Gaussian sum formula](https://en.wikipedia.org/wiki/Gauss_sum)

Comment (Leoriem-code) [reply fr/tri/Tri_par_insertion.md:57]: yes, but we can't make a difference between the worst case and the average one if we only say the complexity is O(N^2)

**Review by Panquesito7 — APPROVED** [review]

**Review by appgurueu — CHANGES_REQUESTED** [review]
I just had a look at the file sizes: There are large GIFs included totaling ~80 MB, which seem to be redundant as there are MP4 files with the same title. Please remove the GIFs and replace them with the MP4s if that's the case.

You should be able to use HTML5 elements to embed videos:
```html
<video controls=""><source src="/path/to/video.mp4" type="video/mp4">Your browser does not support the video tag.</video>
```
Your MP4s appear to be corrupted though (or use an unsupported encoding)? I can't view them on Firefox.

**Comment by Leoriem-code** [comment]
@appgurueu
When I do this, Github doesn't show anything at all.
I tried using the `` syntax but it only showed a link to view the video.  
That is why I used .gif, it worked.
I will look for a solution, but I think it's best to have a link.

**Comment by appgurueu** [comment]
> @appgurueu When I do this, Github doesn't show anything at all. I tried using the `` syntax but it only showed a link to view the video. That is why I used .gif, it worked. I will look for a solution, but I think it's best to have a link.

While GitHub may not support, I hope the website does? Otherwise we should be able to fix it. The `` syntax is for images and can't be used. I agree that links are fine.

**Comment by Leoriem-code** [comment]
I decided to use link rather than trying to fix something that I don't know where to start.

**Comment by appgurueu** [comment]
> I decided to use link rather than trying to fix something that I don't know where to start.

You deleted other GIF files too. Are you sure that you broke no links in the process?

**Comment by Leoriem-code** [comment]
You're right, I let a typo slip. But now it's okay, I made sure of it

**Review by appgurueu — APPROVED** [review]
Looks good to me

**Comment by Panquesito7** [comment]
> > @appgurueu When I do this, Github doesn't show anything at all. I tried using the `` syntax but it only showed a link to view the video. That is why I used .gif, it worked. I will look for a solution, but I think it's best to have a link.
> 
> While GitHub may not support this, I hope the website does? Otherwise we should be able to fix it. The `` syntax is for images and can't be used. I agree that links are fine.

We don't have a website for this. 😅

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍
Thank you for your awesome help, @appgurueu! 🎉

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**